CMakeLists.txt

```diff
@@ -129,7 +133,7 @@ configure_file(${CMAKE_CURRENT_SOURCE_DIR}/cmake/ArborXSettings.cmake.in
 include(CMakePackageConfigHelpers)
 configure_package_config_file(cmake/ArborXConfig.cmake.in
   ${CMAKE_CURRENT_BINARY_DIR}/ArborXConfig.cmake
-  INSTALL_DESTINATION lib/cmake/ArborX
+  INSTALL_DESTINATION ${ARBORX_INSTALL_PACKAGEDIR}
```

Comment:
I would prefer to get rid of the variable and use `"${CMAKE_INSTALL_LIBDIR}/cmake/ArborX"` here and the other places. This way we would not need to do a lookup of what that non-standard variable is actually pointing to.

Reply:
But then it is not configurable, which was the point of this PR.

Reply:
Maybe I misunderstood the original issue. I thought you were saying that it was unconditionally installed in `lib` instead of `lib64`. If that's the case, would not using `CMAKE_INSTALL_LIBDIR` fix that? Or do you need it to be configurable beyond that? If yes, what is the use case?

Reply:
As there is nothing architecture specific in them, I want to be able to install the CMake target file in `${CMAKE_INSTALL_DATADIR}`.

Reply:
Digging a bit more: https://discourse.cmake.org/t/what-should-the-destination-be-for-a-header-only-librarys-cmake-config-file/8473

_"For packages that do not produce binaries (e.g. packages that provide CMake targets that are header-only libraries), it may be better to put the package config files in an architecture-independent location. For this, share/cmake/<packageName> may be more suitable."_

I think ultimately it is your call and I will then make it work on the Fedora side.

If you don't want another CMake variable, then let's do with `${CMAKE_INSTALL_DATADIR}/cmake/ArborX` everywhere and call it a day.

Reply:
Done!